src/unix/fs.c

```diff
@@ -1075,6 +1075,17 @@ static ssize_t uv__fs_sendfile(uv_fs_t* req) {
      */
 
 #if defined(__FreeBSD__) || defined(__DragonFly__)
+#if defined(__FreeBSD__)
+    off_t off;
+
+    off = req->off;
+    r = uv__fs_copy_file_range(in_fd, &off, out_fd, NULL, req->bufsml[0].len, 0);
+    if (r > 0) {
+        r = off - req->off;
+        req->off = r;
+        return r;
+    }
```

Comment:
Falls through to the sendfile() call in the `r == 0` case. Maybe that's harmless but it's still an extra system call so perhaps it shouldn't?

Reply:
fair enough